### PR TITLE
added users orders route with jwt authentication

### PR DESCRIPTION
**Pull request by ElisaTumwebaze**

### CODE REVIEW

File: controllers/userController.js [hunk 1]
```diff
@@ -73,5 +74,29 @@ login_user = async (req,res)=>{
         console.error(err.message)
     }
     }
+    post_orders =verifyToken,(req,res)=>{
```
Comment (katunold) [inline controllers/userController.js]: @ElisaTumwebaze  Kindly add the validation middleware at the route level, to have this organised in a much better way

File: controllers/userController.js [hunk 2]
```diff
@@ -73,5 +74,29 @@ login_user = async (req,res)=>{
...
+        if(typeof bearerHeader !=='undefined'){
+            const bearerToken = bearerHeader.split(' ')[1]
+            req.token = bearerToken;
+            next();
+        }else{
+            res.sendStatus(403);
+        }
+    }
+
+module.exports ={signup_user,login_user,post_orders}
```
Comment (katunold) [inline controllers/userController.js]: Kindly add an empty line character at the end of this file

File: routes/authRoutes.js [hunk 3]
```diff
@@ -4,5 +4,6 @@ const router = express.Router();
 //authentication routes
 router.post('/auth/signup',authController.signup_user);
 router.post('/auth/login',authController.login_user);
+router.post('/users/orders',authController.post_orders);
 
 module.exports = router;
```
Comment (katunold) [inline routes/authRoutes.js]: Kindly add an empty line character at the end of this file

**Review by katunold — CHANGES_REQUESTED** [review]

File: controllers/userController.js (hunk 1 above)
Comment (Luleherll) [reply controllers/userController.js]: @ElisaTumwebaze JWT middleware has to be at a route definition

**Comment by ElisaTumwebaze** [comment]
separated login and signup controllers to make the code easily manageable

**Pull request merged**